src/brsTypes/components/RoArray.ts

```diff
@@ -180,4 +181,21 @@ export class RoArray extends BrsComponent implements BrsValue, BrsIterable {
             return BrsInvalid.Instance;
         },
     });
+
+    private join = new Callable("join", {
+        signature: {
+            args: [new StdlibArgument("separator", ValueKind.String)],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter, separator: BrsString) => {
+            if (
+                this.elements.some(function(element) {
+                    return !(element instanceof BrsString);
+                })
+            ) {
+                throw new Error("roArray.Join: Array contains non-string value(s).");
```

Comment:
Watch out — it looks like the reference BrightScript implementation returns an empty string here.  Executing this file:

```brightscript
sub main()
    a = [1, 2, 3].join(",")
    print a
    print type(a)
end sub
```

results in this output:

```
BRIGHTSCRIPT: ERROR: roArray.Join: non-string value at index 0: pkg:/source/main.brs(3)

String
```

Please update this implementation and add a unit test to ensure that works 😃